native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeImageMojo.java

```diff
@@ -243,7 +243,29 @@ protected List<String> getBuildArgs() throws MojoExecutionException {
 
         if (buildArgs != null && !buildArgs.isEmpty()) {
             for (String buildArg : buildArgs) {
-                cliArgs.addAll(Arrays.asList(buildArg.split("\\s+")));
+                if(buildArg.startsWith("\\Q") ||
+                        buildArg.startsWith("-H:ConfigurationFileDirectories")) {
+                    cliArgs.add(buildArg);
+                    continue;
+                }
+                String[] args = buildArg.split("\\s+");
+                int i=0;
+                while(i < args.length) {
+                    String a =args[i];
+                    if (a.charAt(0) == System.getProperty("user.home").charAt(0)) {
+                        StringBuilder path = new StringBuilder(a);
+                        i++;
+                        while( i< args.length && args[i].toLowerCase().charAt(0) <= 'z' &&
```

Comment:
Is the intent of this to consider only args with letters? Why? Also, what about numbers.

A better option is `Character.isLetterOrDigit()`

Reply:
This is to construct the path that is being split, so I assumed that paths may contain only letters

Reply:
In fact you should check for anything but `File.separatorChar`. Paths may contain numbers, dashes, underscores, dots... almost everything but `File.separatorChar`